core/commands/pin.go

```diff
@@ -187,7 +187,7 @@ Example:
 		cmds.StringArg("ipfs-path", false, true, "Path to object(s) to be listed."),
 	},
 	Options: []cmds.Option{
-		cmds.StringOption("type", "t", "The type of pinned keys to list. Can be \"direct\", \"indirect\", \"recursive\", or \"all\". Defaults to \"recursive\"."),
+		cmds.StringOption("type", "t", "The type of pinned keys to list. Can be \"direct\", \"indirect\", \"recursive\", or \"all\". Defaults to \"all\"."),
```

Comment:
Could we remove the 'Defaults to "All"' from the help string and add `.Default("all")` to the end of the `StringOption` constructor?

The down later where we check `if typeStrFound {` we can drop that check and just rely on the code to pick its own default value correctly.